include/ck_tile/host/arg_parser.hpp

```diff
@@ -12,14 +12,63 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+#include <cassert>
 
 namespace ck_tile {
+struct ValidNumber
+{
+    static bool isNumber(std::string s)
+    {
+        std::unordered_map<std::string, int> dfa[8] = {{{"digit", 1}, {"sign", 2}, {"dot", 3}},
+                                                       {{"digit", 1}, {"dot", 4}, {"exponent", 5}},
+                                                       {{"digit", 1}, {"dot", 3}},
+                                                       {{"digit", 4}},
+                                                       {{"digit", 4}, {"exponent", 5}},
+                                                       {{"sign", 6}, {"digit", 7}},
+                                                       {{"digit", 7}},
+                                                       {{"digit", 7}}};
+        int curr_state                              = 0;
+        std::string group;
+        for(char c : s)
+        {
+            if(std::isdigit(c))
+            {
+                group = "digit";
+            }
+            else if(c == '+' || c == '-')
+            {
+                group = "sign";
+            }
+            else if(c == 'e' || c == 'E')
+            {
+                group = "exponent";
+            }
+            else if(c == '.')
+            {
+                group = "dot";
+            }
+            else
+            {
+                return false;
+            }
+            if(dfa[curr_state].find(group) == dfa[curr_state].end())
+            {
+                return false;
+            }
+            curr_state = dfa[curr_state][group];
+        }
+        return curr_state == 1 || curr_state == 4 || curr_state == 7;
+    }
+};
```

Comment:
I don't think this is really needed. Let's assume that when the user should provide the number then he would do it in a correct way. If the conversion would fail then there should be an error.

Reply:
I don't think so that we're able to predict what the user will do, let alone make assumptions about it. In current scenario if the conversion would fail then you get SIGSEGV in more of those cases, but you don't exactly know where.

Reply:
hey guys, `ck_tile/host/` related utilities, functions is not core part of ck (we will use this by our self, maybe no outside users will use this, cause they only need the core part of our kernels, which is under `ck_tile/ops/`), so the arg_parser here if not that perfect, just let it be, and let's focus more on the kernel....

Reply:
@carlushuang  Sure, it was just only a small addition when the vector support was adding.